### ISSUE TITLE
Incorrect declaration file (import/default)

### ISSUE THREAD

**Issue by pilikan** [prettier/eslint-config-prettier#324]
Eslint-config-prettier has incorrect declaration file: https://github.com/prettier/eslint-config-prettier/pull/294/files (v10.0.3)

Error affects the rule: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/default.md

Result:
`No default export found in imported module "eslint-config-prettier/flat"  import/default`

**Comment by JounQin** [comment]
The module is commonjs, so you should use `import * as eslintConfigPrettiet` instead.

**Comment by JounQin** [comment]
Please provide a minimal but runnable reproduction then.

**Comment by pilikan** [comment]
> Please provide a minimal but runnable reproduction then.

https://stackblitz.com/edit/vitejs-vite-9aq7wlgw?file=package.json,eslint.config.js&terminal=dev

Console command: `npm run lint:js -- --fix`

**Comment by pilikan** [comment]
> Please provide a minimal but runnable reproduction then.

https://stackblitz.com/edit/vitejs-vite-9aq7wlgw?file=package.json,eslint.config.js&terminal=dev

Console command: `npm run lint:js -- --fix`

**Comment by JounQin** [comment]
@pilikan 

So you're using https://github.com/import-js/eslint-import-resolver-typescript together with https://github.com/import-js/eslint-plugin-import, right?

I think this a false postive reported from `eslint-plugin-import`, it should read `package.json` to determine whether `import` statements in `` should be recognized as ESM or commonjs.

cc @SukkaW

But of course, we can change the declaration here:

```ts
// index.d.ts
declare const eslintConfigPrettier: {
  rules: Record<string, 0 | "off">;
};

export = eslintConfigPrettier;

// flat.d.ts
import eslintConfigPrettier from "./index.js";

declare const eslintConfigPrettierFlat: typeof eslintConfigPrettier & {
  name: "config-prettier";
};

export = eslintConfigPrettierFlat;
```

Would you like to raise a PR for it? @pilikan
